src/services/item/plugins/publication/validation/validationQueue.ts

```diff
@@ -0,0 +1,87 @@
+import { hoursToSeconds } from 'date-fns';
+import { Redis } from 'ioredis';
+import { singleton } from 'tsyringe';
+
+import { getIdsFromPath } from '@graasp/sdk';
+
+import { ItemValidationAlreadyExist } from './errors';
+
+const EXPIRES_IN_SECONDS = hoursToSeconds(2);
+const REDIS_PROGRESS_VALIDATION_KEY = 'validation_in_progress';
+const buildRedisProgressKey = (itemId: string) => `${REDIS_PROGRESS_VALIDATION_KEY}:${itemId}`;
+
+/**
+ * Returns the item id of the last item in the path.
+ * @param itemPath The item path to extract the id.
+ * @returns the id of the last item in the path.
+ */
+const getItemIdFromPath = (itemPath: string) => {
+  const itemIds = getIdsFromPath(itemPath);
+  const len = itemIds.length;
+  return itemIds[len - 1];
+};
+
+@singleton()
+export class ValidationQueue {
```

Comment:
Not sure it's a queue, but I can guess that we want to have a real queue at some point. Maybe let's add a comment to explain that at the start of the class?

Reply:
Also since it's called `validation` maybe we don't need to add `validation` in the names of the functions. Especially since they are not real validations? Maybe mention `progress`?

Reply:
Yeah, for now this class only handles the in progress validations, I didn't found a good name, do you have suggestions ?

Reply:
Maybe we can rename the function (again, sorry) like `addInProgressIndication`?